docs/specification.md

```diff
@@ -57,33 +51,83 @@ Compile-time value parameters are captured by the `Config` type (parameterized b
 data Config proposalMetadata = Config
   { cDaoName :: Text
   -- ^ Name of the DAO.
-  , cDaoDescription :: Text
+  , cDaoDescription :: Markdown
   -- ^ Description of the DAO.
-  , cProposalCheck :: Lambda (Natural, Proposal proposalMetadata) Bool
+  , cUnfrozenTokenMetadata :: FA2.TokenMetadata
+  -- ^ FA2 metadata for unfrozen token.
+  , cFrozenTokenMetadata :: FA2.TokenMetadata
+  -- ^ FA2 metadata for frozen token.
+  , cProposalCheck :: forall s
+      . (ProposeParams proposalMetadata) & s
+    :-> Bool & s
   -- ^ A lambda used to verify whether a proposal can be submitted.
   -- It checks 2 things: the proposal itself and the amount of tokens frozen upon submission.
   -- It allows the DAO to reject a proposal by arbitrary logic and captures bond requirements
-  -- (i. e. how much one should "freeze" in order to propose).
-  , cUnfrozenTokenMetadata :: FA2Metadata
-  -- ^ FA2 metadata for unfrozen token.
-  , cFrozenTokenMetadata :: FA2Metadata
-  -- ^ FA2 metadata for frozen token.
-  , cDecisionLambda :: Lambda
-      (Proposal proposalMetadata, Storage proposalMetadata)
-      (List Operation, Storage proposalMetadata)
-  -- ^ The decision lambda is executed based on a successful proposal.
-  , cRejectedProposalReturnValue :: Lambda Natural Natural
+  , cRejectedProposalReturnValue :: forall s
+      . (Proposal proposalMetadata) & s
+    :-> ("slash_amount" :! Natural) & s
   -- ^ When a proposal is rejected, the value that voters get back can be slashed.
   -- This lambda specifies how many tokens will be received.
   -- For example, if Alice freezes 100 tokens to vote and this lambda divides the value by 2,
   -- only 50 tokens will be unfrozen and other 50 tokens will be burnt.
+  , cDecisionLambda :: forall s
+      . (Proposal proposalMetadata) & (Storage proposalMetadata) & s
+    :-> (List Operation, Storage proposalMetadata) & s
+  -- ^ The decision lambda is executed based on a successful proposal.
+  , cMaxProposals :: Natural
+  , cMaxVotes :: Natural
+  , cMaxQuorumThreshold :: Natural
+  , cMinQuorumThreshold :: Natural
+  , cMaxVotingPeriod :: Natural
+  , cMinVotingPeriod :: Natural
```

Comment:
Can you please comment these fields too?

Reply:
I've added comment to these fields.